compiler/qsc_passes/src/capabilitiesck.rs

```diff
@@ -274,6 +277,17 @@ impl<'a> Checker<'a> {
         }
     }
 
+    /// Check that there are no custom measurements in Base Profile.
```

Comment:
Even though this approach will surface an error when someone is using a custom measurement in Base profile, it seems like the pattern used is a bit different from the pattern used for all the other features and capabilities.

The RCA pattern and how errors are surfaced is the following:
- A feature is identified and represented by a new flag in `RuntimeFeatureFlags`.
    - In this case, the new feature could be something like `CustomMeasurement`.
- The feature flag maps to a set of capabilities through the `target_capabilities` function.
    - In this case it would map to `TargetCapabilityFlags::Adaptive`.
- The feature is identified as part of RCA, setting the `CustomMeasurement` flag where appropriate.
- The capabilities check pass automatically maps unsupported features based on the capabilities passed, so in theory no changes would be needed here.

I would suggest to use the established pattern instead. If you have any questions, I'll be happy to help!